### DISCUSSION TITLE
Can I use pre-existing AWS IAM Users/Roles to authenticate to Minio?

### DISCUSSION THREAD

**Discussion by mitchellkager** [minio/minio#19759] [Q&A] [answered]
Let's say I want to access Minio from an AWS Lambda. If I were connecting to S3, I could easily use RBAC to Allow the lambda execution role to contact S3. Assume that me Lambda has network access to Minio -- what is the easiest way to configure Minio's Identity Providers to reference the Lambda's AWS IAM Role as a Principal in my policies?

It seems like I can use any slew of AWS services to support subsets of this problem: LDAP, Keycloak, OpenID. But, these each require individualized reconfiguration, and might require me to set up new auth flows in all of my clients e.g. to integrate a Lambda with OpenID I would have to incorporate Cognito to obtain a Web Identity.

And I can't just register my roles' Access Keys in Minio, since certain Access Keys are prone to rotation.

I want a general solution that will enable me to reference IAM Users & Roles as principals in my Minio policies.

**Comment by harshavardhana** [comment, marked as answer]
AWS Lambda has nothing to do with MinIO. It will never work.

**Reply by mitchellkager** [reply]
so you're saying that there's no way to configure AWS STS/IAM as an External Identity Provider? which would suggest to me that the best way to authenticate from AWS cloud to self-hosted Minio, would be to user hard-coded access keys?
